lib/Backend/arm64/LowerMD.cpp

```diff
@@ -3841,52 +3843,55 @@ LowererMD::GenerateFastNeg(IR::Instr * instrNeg)
 
     bool isInt = (opndSrc1->IsTaggedInt());
 
-    if (opndSrc1->IsRegOpnd() && opndSrc1->AsRegOpnd()->m_sym->m_isNotInt)
+    if (opndSrc1->IsRegOpnd() && opndSrc1->AsRegOpnd()->IsNotInt())
     {
         return true;
     }
 
     labelHelper = IR::LabelInstr::New(Js::OpCode::Label, this->m_func, true);
-
-    // Load src's at the top so we don't have to do it repeatedly.
-    if (!opndSrc1->IsRegOpnd())
-    {
-        opndSrc1 = IR::RegOpnd::New(opndSrc1->GetType(), this->m_func);
-        LowererMD::CreateAssign(opndSrc1, instrNeg->GetSrc1(), instrNeg);
-    }
-
     if (!isInt)
     {
         GenerateSmIntTest(opndSrc1, instrNeg, labelHelper);
     }
 
+    // For 32 bit arithmetic we copy them and set the size of operands to be 32 bits.
+    opndSrc1 = opndSrc1->UseWithNewType(TyInt32, this->m_func);
     GenerateTaggedZeroTest(opndSrc1, instrNeg, labelHelper);
 
-    opndDst = instrNeg->GetDst();
-    if (!opndDst->IsRegOpnd())
+    if (opndSrc1->IsEqual(instrNeg->GetDst()))
     {
-        opndDst = IR::RegOpnd::New(opndDst->GetType(), this->m_func);
+        usingNewDst = true;
+        opndDst = IR::RegOpnd::New(TyInt32, this->m_func);
+    }
+    else
+    {
+        opndDst = instrNeg->GetDst()->UseWithNewType(TyInt32, this->m_func);
     }
 
-    // dst = SUB zr, src
-
-    instr = IR::Instr::New(Js::OpCode::SUB, opndDst, IR::RegOpnd::New(nullptr, RegZR, TyMachReg, this->m_func), opndSrc1, this->m_func);
-    instrNeg->InsertBefore(instr);
-
-    // dst = ADD dst, 2
+    // dst = SUBS zr, src
 
-    instr = IR::Instr::New(Js::OpCode::ADDS, opndDst, opndDst, IR::IntConstOpnd::New(2, TyMachReg, this->m_func), this->m_func);
+    instr = IR::Instr::New(Js::OpCode::SUBS, opndDst, IR::RegOpnd::New(nullptr, RegZR, TyInt32, this->m_func), opndSrc1, this->m_func);
     instrNeg->InsertBefore(instr);
 
     // BVS $helper
 
     instr = IR::BranchInstr::New(Js::OpCode::BVS, labelHelper, this->m_func);
     instrNeg->InsertBefore(instr);
 
-    if (opndDst != instrNeg->GetDst())
+    //
+    // Convert TyInt32 operand, back to TyMachPtr type.
+    //
+    if (TyMachReg != opndDst->GetType())
+    {
+        opndDst = opndDst->UseWithNewType(TyMachPtr, this->m_func);
+    }
+
+    GenerateInt32ToVarConversion(opndDst, instrNeg);
+
+    if (usingNewDst)
     {
-        //Now store the result.
-        LowererMD::CreateAssign(instrNeg->GetDst(), opndDst, instrNeg);
+        instr = IR::Instr::New(Js::OpCode::MOV, instrNeg->GetDst(), opndDst, this->m_func);
```

Comment:
>instr [](start = 8, length = 5)

Why removed  LowererMD::CreateAssign? that have size and writebarrier check which is good.

Reply:
I initially made it as similar as possible to the amd64 version. I can change this back to CreateAssign.